library/core/src/intrinsics.rs

```diff
@@ -991,14 +994,14 @@ extern "rust-intrinsic" {
     ///
     /// let v_clone = v_orig.clone();
     ///
-    /// // The no-copy, unsafe way, still using transmute, but not relying on the data layout.
-    /// // Like the first approach, this reuses the `Vec` internals.
-    /// // Therefore, the new inner type must have the
-    /// // exact same size, *and the same alignment*, as the old type.
-    /// // The same caveats exist for this method as transmute, for
-    /// // the original inner type (`&i32`) to the converted inner type
-    /// // (`Option<&i32>`), so read the nomicon pages linked above and also
-    /// // consult the [`from_raw_parts`] documentation.
+    /// // To transmute the inner type of the contents of a container, you must make sure
```

Comment:
I don't think we want to be so general here - in theory, the container could be relying on the type being the same (e.g., to do a TypeId-based lookup in some global map).

Reply:
I now make a reference to the container's *invariants*, which for `Vec` happen to be size and alignment. I also added your `TypeId` example, because I think it's a good idea to warn the users about what a container could be relying on.